File: app/src/main/java/me/zhanghai/android/douya/account/info/AccountContract.java
/*
 * Copyright (c) 2015 Zhang Hai <Dreaming.in.Code.ZH@Gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.douya.account.info;

import me.zhanghai.android.douya.BuildConfig;

public class AccountContract {

    public static final String ACCOUNT_TYPE = BuildConfig.APPLICATION_ID;

    private static final String SUFFIX_API_V2 = ".api_v2";
    private static final String SUFFIX_FRODO = ".frodo";

    public static final String AUTH_TOKEN_TYPE_API_V2 = BuildConfig.APPLICATION_ID + SUFFIX_API_V2;
    public static final String AUTH_TOKEN_TYPE_FRODO = BuildConfig.APPLICATION_ID + SUFFIX_FRODO;

    public static final String KEY_USER_NAME = BuildConfig.APPLICATION_ID + ".user_name";
    public static final String KEY_USER_ID = BuildConfig.APPLICATION_ID + ".user_id";
    public static final long INVALID_USER_ID = -1;
    public static final String KEY_REFRESH_TOKEN_API_V2 = BuildConfig.APPLICATION_ID
            + ".refresh_token" + SUFFIX_API_V2;
    public static final String KEY_REFRESH_TOKEN_FRODO = BuildConfig.APPLICATION_ID
            + ".refresh_token" + SUFFIX_FRODO;
    public static final String KEY_USER_INFO = BuildConfig.APPLICATION_ID + ".user_info";

    private AccountContract() {}
}

File: app/src/main/java/me/zhanghai/android/douya/account/info/AccountContract.kt
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.douya.account.info

import me.zhanghai.android.douya.BuildConfig

object AccountContract {
    const val ACCOUNT_TYPE = BuildConfig.APPLICATION_ID

    const val AUTH_TOKEN_TYPE = "${BuildConfig.APPLICATION_ID}.auth_token.FRODO"

    const val USER_DATA_KEY_USER_ID = "${BuildConfig.APPLICATION_ID}.user_data.USER_ID"

    const val USER_DATA_KEY_USER_NAME = "${BuildConfig.APPLICATION_ID}.user_data.USER_NAME"

    const val USER_DATA_KEY_REFRESH_TOKEN = "${BuildConfig.APPLICATION_ID}.user_data.REFRESH_TOKEN"
}
